.evergreen.yml

```diff
@@ -204,6 +215,97 @@ functions:
+        silent: true
+        script: |
+          set -x
+          set -v
+          set -e
+          cat <<EOF > $(pwd)/aws_e2e_setup.json
+          {
+            "iam_auth_assume_aws_account": "${iam_auth_assume_aws_account}",
+            "iam_auth_assume_aws_secret_access_key": "${iam_auth_assume_aws_secret_access_key}",
+            "iam_auth_assume_role_name":"${iam_auth_assume_role_name}"
+          }
+          EOF
+          export AWS_ASSUMED_ROLE=${iam_auth_assume_role_name}
+  "setup-aws-auth-test-with-assume-role-credentials":
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: src/github.com/mongodb/mongo-tools-common
+        silent: true
+        script: |
+          set -x
+          set -v
+          set -e
+          # The aws_e2e_assume_role script requires python3 with boto3.
+          venv='venv'
+          if [ -f "$venv"/bin/activate ]; then
+            echo 'using existing virtualenv'
+            . "$venv"/bin/activate
+          elif [ -f "$venv"/Scripts/activate ]; then
+            echo 'using existing virtualenv'
+            . "$venv"/Scripts/activate
+          elif virtualenv "$venv" || python3 -m virtualenv "$venv"; then
+            echo 'creating new virtualenv'
+            if [ -f "$venv"/bin/activate ]; then
+              . "$venv"/bin/activate
+            elif [ -f "$venv"/bin/activate ]; then
+              . "$venv"/Scripts/activate
+            fi
+            if [ "Windows_NT" = "$OS" ]; then
+              pip install boto3
+            else
+              python3 -m pip install --user boto3
+            fi
+          fi
+          pip list
+          ./bin/mongo --nodb aws_e2e_assume_role.js
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: src/github.com/mongodb/mongo-tools-common
+        silent: true
+        script: |
+          set -x
+          set -v
+          set -e
+          cat <<'EOF' > "$(pwd)/prepare_mongodb_aws.sh"
+            alias urlencode='python -c "import sys, urllib as ul; sys.stdout.write(ul.quote_plus(sys.argv[1]))"'
+            alias jsonkey='python -c "import json,sys;sys.stdout.write(json.load(sys.stdin)[sys.argv[1]])" < $(pwd)/creds.json'
+            USER=$(jsonkey AccessKeyId)
+            USER=$(urlencode $USER)
+            PASS=$(jsonkey SecretAccessKey)
+            PASS=$(urlencode $PASS)
+            SESSION_TOKEN=$(jsonkey SessionToken)
+            SESSION_TOKEN=$(urlencode $SESSION_TOKEN)
+            MONGODB_URI="mongodb://$USER:$PASS@localhost"
+          EOF
+  "run-aws-auth-test-with-assume-role-credentials":
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: src/github.com/mongodb/mongo-tools-common
+        script: |
+          set -x
+          set -v
+          . ./set_goenv.sh
+          GOROOT="" set_goenv || exit
+          MONGO_ARGS="${mongo_args}"
+          if [ "${AWS_AUTH}" = "true" ]; then
```

Comment:
Is AWS_AUTH var necessary here? Since the task name `run-aws-auth-test-with-assume-role-credentials` specifically means we will run aws auth here, maybe we can just hardcode
``` export TOOLS_TESTING_AWS_AUTH="true"``` 
?

Reply:
Good catch! This if statement is just an artifact of when I combined the Kerberos and AWS evergreen task; I no longer need it now that the tasks are once again separated.